Build failure: lua-language-server-3.13.4

## Steps To Reproduce

Steps to reproduce the behavior:

1. build lua-language-server-3.13.4

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
> bee.lua/3rd/fmt/fmt/color.h:578:13: error: no member named 'copy' in namespace 'std'; did you mean 'bcopy'?
       >   578 |       out = std::copy(emphasis.begin(), emphasis.end(), out);
       >       |             ^~~~~~~~~
       >       |             bcopy
       > /nix/store/2y71rl5mnazwnpwpcq07xx97jl7sy5lh-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/strings.h:71:7: note: 'bcopy' declared here
       >    71 | void     bcopy(const void *, void *, size_t) __POSIX_C_DEPRECATED(200112L);
       >       |          ^
       > In file included from bee.lua/bee/subprocess/subprocess_posix.cpp:2:
       > In file included from bee.lua/bee/nonstd/format.h:36:
       > In file included from bee.lua/3rd/fmt/fmt/xchar.h:15:
       > bee.lua/3rd/fmt/fmt/color.h:584:13: error: no member named 'copy' in namespace 'std'; did you mean 'bcopy'?
       >   584 |       out = std::copy(foreground.begin(), foreground.end(), out);
       >       |             ^~~~~~~~~
       >       |             bcopy
       > /nix/store/2y71rl5mnazwnpwpcq07xx97jl7sy5lh-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/strings.h:71:7: note: 'bcopy' declared here
       >    71 | void     bcopy(const void *, void *, size_t) __POSIX_C_DEPRECATED(200112L);
       >       |          ^
       > In file included from bee.lua/bee/subprocess/subprocess_posix.cpp:2:
       > In file included from bee.lua/bee/nonstd/format.h:36:
       > In file included from bee.lua/3rd/fmt/fmt/xchar.h:15:
       > bee.lua/3rd/fmt/fmt/color.h:590:13: error: no member named 'copy' in namespace 'std'; did you mean 'bcopy'?
       >   590 |       out = std::copy(background.begin(), background.end(), out);
       >       |             ^~~~~~~~~
       >       |             bcopy
       > /nix/store/2y71rl5mnazwnpwpcq07xx97jl7sy5lh-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/strings.h:71:7: note: 'bcopy' declared here
       >    71 | void     bcopy(const void *, void *, size_t) __POSIX_C_DEPRECATED(200112L);
       >       |          ^
       > In file included from bee.lua/bee/subprocess/subprocess_posix.cpp:2:
       > In file included from bee.lua/bee/nonstd/format.h:36:
       > In file included from bee.lua/3rd/fmt/fmt/xchar.h:15:
       > bee.lua/3rd/fmt/fmt/color.h:595:13: error: no member named 'copy' in namespace 'std'; did you mean 'bcopy'?
       >   595 |       out = std::copy(reset_color.begin(), reset_color.end(), out);
       >       |             ^~~~~~~~~
       >       |             bcopy
       > /nix/store/2y71rl5mnazwnpwpcq07xx97jl7sy5lh-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/strings.h:71:7: note: 'bcopy' declared here
       >    71 | void     bcopy(const void *, void *, size_t) __POSIX_C_DEPRECATED(200112L);
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
- system: `"aarch64-darwin"`
 - host os: `Darwin 24.2.0, macOS 15.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.25.3`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/73cnf2k7mji95fcflvay15pdw4skhhk5-source`
 - 
## Notify maintainers
@figsoda @gepbird @sei40kr

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/lu/lua-language-server/package.nix
+++ b/pkgs/by-name/lu/lua-language-server/package.nix
@@ -8,8 +8,7 @@
   makeWrapper,
 
   # buildInputs
-  apple-sdk_11,
-  darwinMinVersionHook,
+  fmt,
   rsync,
 
   versionCheckHook,
@@ -33,12 +32,15 @@ stdenv.mkDerivation (finalAttrs: {
     makeWrapper
   ];
 
-  buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [
-    # aligned_alloc
-    apple-sdk_11
-    (darwinMinVersionHook "10.15")
-    rsync
-  ];
+  buildInputs =
+    [
+      fmt
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      rsync
+    ];
+
+  env.NIX_LDFLAGS = "-lfmt";
 
   postPatch =
     ''
@@ -46,6 +48,15 @@ stdenv.mkDerivation (finalAttrs: {
       # this feature is not used in lua-language-server
       substituteInPlace 3rd/bee.lua/test/test.lua \
         --replace-fail 'require "test_filewatch"' ""
+
+      # use nixpkgs fmt library
+      for d in 3rd/bee.lua 3rd/luamake/bee.lua
+      do
+        rm -r $d/3rd/fmt/*
+        touch $d/3rd/fmt/format.cc
+        substituteInPlace $d/bee/nonstd/format.h $d/bee/nonstd/print.h \
+          --replace-fail "include <3rd/fmt/fmt" "include <fmt"
+      done
 
       # flaky tests on linux
       # https://github.com/LuaLS/lua-language-server/issues/2926

